src/java.base/share/classes/sun/security/provider/certpath/DistributionPointFetcher.java

```diff
@@ -101,16 +101,24 @@ public static Collection<X509CRL> getCRLs(X509CRLSelector selector,
                 }
                 return Collections.emptySet();
             }
-            List<DistributionPoint> points =
-                    ext.getDistributionPoints();
+            List<DistributionPoint> points = ext.getDistributionPoints();
             Set<X509CRL> results = new HashSet<>();
+            CertStoreException savedCSE = null;
             for (Iterator<DistributionPoint> t = points.iterator();
                  t.hasNext() && !Arrays.equals(reasonsMask, ALL_REASONS); ) {
-                DistributionPoint point = t.next();
-                Collection<X509CRL> crls = getCRLs(selector, certImpl,
-                    point, reasonsMask, signFlag, prevKey, prevCert, provider,
-                    certStores, trustAnchors, validity, variant, anchor);
-                results.addAll(crls);
+                try {
+                    DistributionPoint point = t.next();
+                    Collection<X509CRL> crls = getCRLs(selector, certImpl,
+                        point, reasonsMask, signFlag, prevKey, prevCert, provider,
+                        certStores, trustAnchors, validity, variant, anchor);
+                    results.addAll(crls);
+                } catch (CertStoreException cse) {
+                    savedCSE = cse;
```

Comment:
Yes, that would be a good idea to call `addSupressed` in case there is more than one exception. For debug, there is already a debug statement further down the call stack in either the URICertStore or LDAPCertStore code where the exception cause is captured so I didn't want to duplicate that info.